iceoryx_hoofs/platform/win/source/time.cpp

```diff
@@ -162,12 +162,23 @@ int timer_getoverrun(timer_t)
 
 int clock_gettime(clockid_t clk_id, struct timespec* tp)
 {
-    if (clk_id != CLOCK_REALTIME)
+    if (clk_id == CLOCK_MONOTONIC)
     {
-        fprintf(stderr, "Windows Version of clock_gettime supports only CLOCK_REALTIME clockID\n");
+        constexpr double NANO_SECONDS_PER_SECOND = 1000000000.0;
+        LARGE_INTEGER ticksPerSecond, ticks;
+        QueryPerformanceFrequency(&ticksPerSecond);
+        QueryPerformanceCounter(&ticks);
+        double seconds = static_cast<double>(ticks.QuadPart) / static_cast<double>(ticksPerSecond.QuadPart);
+        tp->tv_sec = static_cast<time_t>(seconds);
+        tp->tv_nsec = static_cast<long>((seconds - std::floor(seconds)) * NANO_SECONDS_PER_SECOND);
```

Comment:
can't we just use `chrono::steady_clock`?

Reply:
I decided against it.
1. With msvc 2015 the steady clock is implemented exactly in this way like I implemented it above.
2. https://docs.microsoft.com/en-us/cpp/standard-library/chrono?view=msvc-160 mentions that different implementations of `steady_clock` exists and I wanted to have a uniform implementation here in windows. Not that someone uses clang or something else and then the steady clock does not fulfill the requirements of steadiness and monotonicity.
3. I have no idea if we ever want to support older msvc versions, then the steady_clock requirements are violated, but not with this approach.

I am a little bit intimidated by this and I felt this would be the safe approach.

Reply:
@elfenpiff 
I think it's more safe that we only support msvc compiler version upwards from version 1900 which is used in VS 2015.
We mainly do the windows support for ROS 2 where the in the recent releases mainly VS 2019 is supported.
I vote for a [compile-time check](https://cmake.org/cmake/help/v3.10/variable/MSVC_VERSION.html) to prevent building with older versions and removing this custom implementation here.

Reply:
I've also a slight tendency towards what dkroenke says.

Reply:
done